From 7e4aac131868a12fc8b222fd59f3f2cc3bf71d3f Mon Sep 17 00:00:00 2001
From: Frode Isaksen <frodex.isaksen@intel.com>
Date: Fri, 6 Mar 2015 18:02:23 +0100
Subject: [PATCH] bcmdhd: PCIe driver is not using OOB interrupts..

..so do not request the GPIO to be used as IRQ.

Change-Id: Ie5bad55495dd3f6100fa9b9360dce10beb98f2f4
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1164
Signed-off-by: Frode Isaksen <frodex.isaksen@intel.com>
---
 drivers/net/wireless/bcmdhd/dhd_linux_platdev.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/net/wireless/bcmdhd/dhd_linux_platdev.c b/drivers/net/wireless/bcmdhd/dhd_linux_platdev.c
index 471c8b6..1edda83 100644
--- a/drivers/net/wireless/bcmdhd/dhd_linux_platdev.c
+++ b/drivers/net/wireless/bcmdhd/dhd_linux_platdev.c
@@ -260,7 +260,6 @@ static int wifi_plat_dev_drv_probe_acpi(struct platform_device *pdev)
 	wifi_adapter_info_t *adapter;
 	acpi_handle handle;
 	struct acpi_device *adev;
-	int gpio_num = -1;
 	int irq_num = -1;
 
 	/* Android style wifi platform data device ("bcmdhd_wlan" or "bcm4329_wlan")
@@ -287,12 +286,14 @@ static int wifi_plat_dev_drv_probe_acpi(struct platform_device *pdev)
 			/* Ignore SDH SDIO controller ACPI PM state */
 			adev->power.flags.ignore_parent = 1;
 		}
-		gpio_num = desc_to_gpio(gpiod_get_index(&pdev->dev, NULL, 0));
-		pr_err("%s: Using ACPI table to get GPIO number: %d\n", __FUNCTION__, gpio_num);
-		if (gpio_num > 0) {
-			irq_num = gpio_to_irq(gpio_num);
+#if defined(OOB_INTR_ONLY)
+		irq_num = desc_to_gpio(gpiod_get_index(&pdev->dev, NULL, 0));
+		pr_err("%s: Using ACPI table to get GPIO number: %d\n", __FUNCTION__, irq_num);
+		if (irq_num > 0) {
+			irq_num = gpio_to_irq(irq_num);
 			pr_err("%s: IRQ number: %d\n", __FUNCTION__, irq_num);
 		}
+#endif
 	} else {
 		DHD_ERROR(("%s: Null ACPI_HANDLE, try legacy probe\n", __FUNCTION__));
 		return wifi_plat_dev_drv_probe(pdev);
-- 
1.9.1

